google_auth_oauthlib/flow.py

```diff
@@ -160,7 +159,9 @@ def from_client_config(cls, client_config, scopes, **kwargs):
                 client_config, scopes, **kwargs))
 
         redirect_uri = kwargs.get('redirect_uri', None)
-        return cls(session, client_type, client_config, redirect_uri)
+        code_verifier = kwargs.get('code_verifier', None)
+
+        return cls(session, client_type, client_config, redirect_uri, code_verifier)
```

Comment:
This will fail because few rows higher you pass `kwargs` to `google_auth_oauthlib.helpers.session_from_client_config` which in turns pass those to `requests_oauthlib.OAuth2Session` which will not accept a `code_verifier` argument.

You need to pop the `code_verifier` from kwargs before a session is created. Have a look at my submitted PR https://github.com/googleapis/google-auth-library-python-oauthlib/pull/47

Reply:
Thank you @tamaroth!